test/jdk/javax/xml/jaxp/datatype/8033980/SerializationTest.java

```diff
@@ -25,160 +25,314 @@
 
-    }
+            // Create the Data for JDK under test.
+            gregorianCalendarAndDurationSerData[0] = new GregorianCalendarAndDurationSerData() {
+                @Override
+                public byte[] getGregorianCalendarByteArray() {
+                    return baos.toByteArray();
+                }
 
-    public void testReadCal() {
-        try {
-            for (String javaVersion : JDK) {
-                XMLGregorianCalendar d1 = (XMLGregorianCalendar) fromFile(
-                        javaVersion + FILENAME_CAL);
-                if (!d1.toString().equalsIgnoreCase(EXPECTED_CAL)) {
-                    fail("Java version: " + javaVersion
-                            + "\nExpected: " + EXPECTED_CAL
-                            + "\nActual: " + d1.toString());
-                } else {
-                    success("testReadCal: read " + javaVersion + " serialized file, passed.");
+                @Override
+                public String getGregorianCalendarBase64() {
+                    return base64;
                 }
-            }
-        } catch (ClassNotFoundException ex) {
-            fail("testReadCal: " + ex.getMessage());
-        } catch (IOException ex) {
-            fail("testReadCal: " + ex.getMessage());
-        }
-    }
 
-    public void testReadDuration() {
-        try {
-            for (String javaVersion : JDK) {
-                Duration d1 = (Duration) fromFile(
-                        javaVersion + FILENAME_DURATION);
-                if (!d1.toString().equalsIgnoreCase(EXPECTED_DURATION)) {
-                    fail("Java version: " + javaVersion
-                            + "\nExpected: " + EXPECTED_DURATION
-                            + "\nActual: " + d1.toString());
-                } else {
-                    success("testReadDuration: read " + javaVersion + " serialized file, passed.");
+                @Override
+                public byte[] getDurationBytes() {
+                    return baos2.toByteArray();
                 }
-            }
-        } catch (ClassNotFoundException ex) {
-            fail("testReadDuration: " + ex.getMessage());
-        } catch (IOException ex) {
-            fail("testReadDuration: " + ex.getMessage());
+
+                @Override
+                public String getDurationBase64() {
+                    return base64dur;
+                }
+            };
+            // To create the pseudocode for <JDK version>GregorianCalendarAndDurationSerData.java for specific version
+            // of JDK (other than 6, 7, 8 and 9), execute this test with that specific JDK version after uncommenting
+            // the below methods calls :
+
+            // generatePseudoCodeForGregCalSerBytes(baos);
+            // generatePseudoCodeForGregCalSerBytesAsBase64(base64);
+            // generatePseudoCodeForDurationSerBytes(baos2);
+            // generatePseudoCodeForDurationSerBytesAsBase64(base64dur);
         }
```

Comment:
I really do not like this being hidden in the setup method

I still would look to have a method that actually creates the classes.   One example of this:

`open/test/jdk/java/io/Serializable/failureAtomicity/FailureAtomicity.java` and the `createSrc` method

The intent is to make this easier to update/enhance going forward. I would probably have. separate utility class/template which can be run to create the saved serialization classes.

SerializationTest would have an introductory comment outlining how to add additional serialized at a via the utility class.